core/serialize/serialize.h

```diff
@@ -24,6 +28,9 @@ class Serializer {
     // Serializes an AST and file hash.
     static std::vector<uint8_t> storeTree(const core::File &file, const ast::ParsedFile &tree);
 
+    // Augment a global state with the name table stored in the cache.
+    static void loadNameTable(GlobalState &gs, const uint8_t *const data);
+
     static void loadGlobalState(GlobalState &gs, const uint8_t *const data);
```

Comment:
If I'm not mistaken, this function is now only used

- in unit tests
- in the part of the code that parses the binary data in Sorbet's payload

I wonder if we should make this return a new GlobalState, instead of populating into an existing one?

That might make it a little less likely that we do something like read the name table into an empty GlobalState and then call this function second, overwriting what we'd read.

Reply:
If we don't want to make `loadGlobalState` return a GlobalState, I think we should probably rename `loadNameTable` to something like `loadAndOverwriteNameTable` for clarity.

Reply:
I think both changes would be good!